xtra-libp2p-ping/src/ping.rs

```diff
@@ -66,6 +67,18 @@ impl xtra::Actor for Actor {
 
         self.spawner = Some(spawner::Actor::new().create(None).spawn(&mut self.tasks));
 
+        match self.endpoint.send(GetConnectionStats).await {
+            Ok(connection_stats) => self
+                .connected_peers
+                .extend(connection_stats.connected_peers),
+            Err(e) => {
+                tracing::error!(
+                    "Unable to receive connection stats from the endpoint upon startup: {e:#}"
+                );
+                ctx.stop();
```

Comment:
Will the actor be immediately restarted? 

If so, and for some reason `GetConnectionStats` keeps erroring, we will bombard the `endpoint` with `GetConnectionStats` requests until our CPU starts to glow 😅

Reply:
We had this consideration when discussing the implementation. The `endpoint` is designed to be always available. It is supervised. we should *always* be able to query `GetConnectionStats` from the `endpoint. 

I am happy to add a `sleep` there to adhere to defensive coding. I don't think we can do much else unless we want to change the design more radically. The protocol actors are designed to be "always available" as well, i.e. they are also supervised.
Would a sleep before restarting the actor be OK for you?

Reply:
> We could actually make the sleep part of the restart policy.

I take it back - we might not want that. This would mean that we would `sleep` in all cases before restarting the actor. I will introduce a specific `sleep` for this specific case.

Reply:
It was more a question but I think the sleep is ok.

Reply:
> > We could actually make the sleep part of the restart policy.
> 
> I take it back - we might not want that. This would mean that we would `sleep` in all cases before restarting the actor. I will introduce a specific `sleep` for this specific case.

it might still be part of the restart policy (e.g. if an actor got stopped with error `A`, start it again straight away, but if with the error `B`, then sleep).

(something to think about for the future)

Reply:
Good point with the  custom error, I think that would be a bit more explicit. Given that we have to implement it for both actors I don't see much advantage there though, so I'm merging. If we have other error cases for these actors that require a sleep we can extract it into the policy.